lib/ansible/modules/network/nxos/nxos_nxapi.py

```diff
@@ -171,6 +171,18 @@ def needs_update(x):
         if want['state'] == 'absent':
             return ['no feature nxapi']
         commands.append('feature nxapi')
+    elif want['state'] == 'absent':
+        return commands
+
+    # The latest version (9.2) of NXOS software has changed the platform default
+    # nxapi transfer to https. Enforce the default nxapi http behavior unless
+    # explicitly asking for https.
+
+    if want.get('https') is None or want.get('https') is False:
```

Comment:
I'm not really sure why you are going through all this rather than just setting the `default` key in `argument_spec`... is there something I'm not seeing that is why the defaults aren't set in this module?

Reply:
Great suggestion.  Refactoring the code now.